pytest_django/fixtures.py

```diff
@@ -7,71 +7,106 @@
 import pytest
 
 from . import live_server_helper
-from .db_reuse import (monkey_patch_creation_for_db_reuse,
-                       monkey_patch_creation_for_db_suffix)
+
 from .django_compat import is_django_unittest
+
 from .lazy_django import get_django_version, skip_if_no_django
 
-__all__ = ['_django_db_setup', 'db', 'transactional_db', 'admin_user',
+__all__ = ['django_db_setup', 'db', 'transactional_db', 'admin_user',
            'django_user_model', 'django_username_field',
            'client', 'admin_client', 'rf', 'settings', 'live_server',
            '_live_server_helper']
 
 
-# ############### Internal Fixtures ################
-
 @pytest.fixture(scope='session')
-def _django_db_setup(request,
-                     _django_test_environment,
-                     _django_cursor_wrapper):
-    """Session-wide database setup, internal to pytest-django"""
+def django_db_modify_db_settings(request):
```

Comment:
This only does the xdist suffix handling, right?
Would be good to factor `django_db_test_name` out of it (where it would only build the test name), and then have this do the xdist related change only - which could also be its own fixture though.
In the end it would be good if you could override a fixture that gets a test_name and return your modified version.